tests/functions/test_le_join.py

```diff
@@ -0,0 +1,485 @@
+    """Raise ValueError if `right` is not a named Series"""
+    sequence = pd.Series(sequence)
+    with pytest.raises(
+        ValueError,
+        match="Unnamed Series are not supported for non-equi joins.",
+    ):
+        left_df.le_join(sequence, "col_a", "col_a")
+
+
+def test_wrong_type_order_by_appearance(left_df, right_df):
+    """Raise TypeError if wrong type is provided for `order_by_appearance`."""
+    with pytest.raises(TypeError):
+        left_df.le_join(right_df, "col_a", "col_a", order_by_appearance="True")
+
+
+def test_wrong_column_presence_right(left_df, right_df):
+    """Raise ValueError if column is not found in `right`."""
+    with pytest.raises(ValueError):
+        left_df.le_join(right_df, "col_a", "col_b")
+
+
+def test_wrong_column_presence_df(left_df, right_df):
+    """Raise ValueError if column is not found in `df`."""
+    with pytest.raises(ValueError):
+        left_df.le_join(right_df, "col_c", "col_a")
+
+
+def test_wrong_column_type_df(left_df, right_df):
+    """Raise ValueError if wrong type is provided for column."""
+    with pytest.raises(TypeError):
+        left_df.le_join(right_df, 1, "col_a")
+        left_df.le_join(right_df, "col_a", 2)
+
+
+def test_wrong_type_suffixes(left_df, right_df):
+    """Raise TypeError if `suffixes` is not a tuple."""
+    with pytest.raises(TypeError):
+        left_df.le_join(right_df, "col_a", "col_a", suffixes=None)
+
+
+def test_wrong_length_suffixes(left_df, right_df):
+    """Raise ValueError if `suffixes` length != 2."""
+    with pytest.raises(
+        ValueError, match="`suffixes` argument must be a 2-length tuple"
+    ):
+        left_df.le_join(right_df, "col_a", "col_a", suffixes=("_x",))
+
+
+def test_suffixes_None(left_df, right_df):
+    """Raise ValueError if `suffixes` is (None, None)."""
+    with pytest.raises(
+        ValueError, match="At least one of the suffixes should be non-null."
+    ):
+        left_df.le_join(right_df, "col_a", "col_a", suffixes=(None, None))
+
+
+def test_wrong_type_suffix(left_df, right_df):
+    """
+    Raise TypeError if one of the `suffixes`
+    is not None or a string type.
+    """
+    with pytest.raises(TypeError):
+        left_df.le_join(right_df, "col_a", "col_a", suffixes=("_x", 1))
+
+
+def test_suffix_already_exists_df(left_df, right_df):
+    """Raise ValueError if label with suffix already exists."""
+    left_df["col_a_x"] = 2
+    with pytest.raises(ValueError):
+        left_df.le_join(right_df, "col_a", "col_a")
+
+
+def test_suffix_already_exists_right(left_df, right_df):
+    """Raise ValueError if label with suffix already exists."""
+    right_df["col_a_y"] = 2
+    with pytest.raises(ValueError):
+        left_df.le_join(right_df, "col_a", "col_a")
+
+
+def test_column_same_type(left_df, right_df):
+    """Raise ValueError if both columns are not of the same type."""
+    with pytest.raises(ValueError):
+        left_df.le_join(right_df, "col_a", "col_c")
+
+
+various = [
```

Comment:
@samukweku I'm wondering what you think about this next idea: What do you think about using `hypothesis` to generate a few categories of dataframes instead of manually specifying them here?

Reply:
@ericmjl ; yes, I think this is a good opportunity for me to learn about hypothesis. However, I'm still thinking about the entire conditional join (speed wise, and API)

Reply:
Understood, got it @samukweku! No wonder you marked this WIP again :smile: (I just noticed). Okie dokes, I'll refrain from taking action here until you feel ready!